var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -48,14 +49,66 @@ class Hip(CMakePackage):
     # See https://github.com/ROCm-Developer-Tools/HIP/pull/2141
     patch('0002-Fix-detection-of-HIP_CLANG_ROOT.patch', when='@3.5.0:')
 
+    def get_hip_path_info(self, env, dependent_spec):
+        # Please note that there is currently a bug in how spack detects hip.
+        # There is a workaound involving some manual changes to the following
+        # lines to hard-code the fallback_path. Manual edits will also be
+        # required in the packages.yaml file. Please contact a developer
+        # for details.
+        mydict = {}
+        fallback_path = os.path.abspath('/opt/rocm')
+        found_fallback_path = os.path.isdir(fallback_path)
```

Comment:
Are rocm/hip installations consistent enough in their structure that we could express the fallback path in terms of relative paths from the hip prefix?

Reply:
If it's an external install that hasn't been messed with by the distro, then yes.  (I don't know of any distros that still mess with this BTW, but fedora used to)

Reply:
Ok, sounds good then. I made the change and added a comment